tests/upgrade/common.sh

```diff
@@ -58,13 +59,95 @@ function docker_compose_down () {
 # Clean environment
 function clean_env () {
     rm -rf node_configs
+    rm -f $FNAME_TXHASHES
 }
 
 # Run command using local generated keys from node_configs/client
-function local_client_exec () {
-    cheqd_noded_docker "$@" --home node_configs/client/.cheqdnode/ --keyring-backend test
+function local_client_tx () {
+    result="$(cheqd_noded_docker $@ --home node_configs/client/.cheqdnode/ --keyring-backend test)"
 }
 
 function make_777 () {
     sudo chmod -R 777 node_configs
-}
\ No newline at end of file
+}
+
+
+
+# Transaction related funcs
+
+function random_string() {
+  echo $RANDOM | base64 | head -c 20
+  return 0
+}
+
+function get_addresses () {
+    local_client_tx keys list
+    addresses=( $(echo "$result" | grep -o 'cheqd1.*') )
+    echo "${addresses[@]}"
+}
+
+CHEQ_AMOUNT="1ncheq"
+
+# Send tokens from the first address in the list to another one
+function send_tokens() {
+    get_addresses 
+
+    OP0_ADDRESS=${addresses[0]}
+    OP1_ADDRESS=${addresses[1]}
+    OP2_ADDRESS=${addresses[2]}
+    OP3_ADDRESS=${addresses[3]}
+
+    local_client_tx tx \
+                    bank \
+                    send $OP0_ADDRESS $OP1_ADDRESS $CHEQ_AMOUNT \
+                    --gas auto \
+                    --gas-adjustment 1.2 \
+                    --gas-prices "25ncheq" \
+                    --chain-id $CHAIN_ID \
+                    -y
+    txhash=$(echo $result | jq ".txhash" | tr -d '"')
+    echo $txhash >> $FNAME_TXHASHES
+}
+
+
+function send_did () {
+
+    # Generate Alice identity key
+    ALICE_VER_KEY="$(cheqd_noded_docker debug ed25519 random)"
+    ALICE_VER_PUB_BASE_64=$(echo "${ALICE_VER_KEY}" | jq -r ".pub_key_base_64")
+    ALICE_VER_PRIV_BASE_64=$(echo "${ALICE_VER_KEY}" | jq -r ".priv_key_base_64")
+    ALICE_VER_PUB_MULTIBASE_58=$(cheqd_noded_docker debug encoding base64-multibase58 "${ALICE_VER_PUB_BASE_64}")
+
+    # Build CreateDid message
+    DID="did:cheqd:testnet:$(random_string)"
+    KEY_ID="${DID}#key1"
+
+    MSG_CREATE_DID='{"id":"'${DID}'","verification_method":[{"id":"'${KEY_ID}'","type":"Ed25519VerificationKey2020","controller":"'${DID}'","public_key_multibase":"'${ALICE_VER_PUB_MULTIBASE_58}'"}],"authentication":["'${KEY_ID}'"]}'
+
+    # Post the message
+    local_client_tx tx cheqd create-did ${MSG_CREATE_DID} ${KEY_ID} \
+        --ver-key "${ALICE_VER_PRIV_BASE_64}" \
+        --from operator0 \
+        --gas-prices "25ncheq" \
+        --chain-id $CHAIN_ID \
+        --output json \
+        -y
+
+
+    txhash=$(echo $result | jq ".txhash" | tr -d '"')
+    echo $txhash >> $FNAME_TXHASHES
```

Comment:
Could you please check transactions using query? TxHash is a good way too, let's keep it. But I'm not sure about TxHash after pool upgrade with migration